ion-c-sys/src/reader.rs

```diff
@@ -510,6 +512,10 @@ impl<'a> IonCReader for IonCReaderHandle<'a> {
     fn read_bytes(&mut self) -> IonCResult<Vec<u8>> {
         let mut len = 0;
         ionc!(ion_reader_get_lob_size(self.reader, &mut len))?;
+        if len == 0 {
+            // short-circuit the empty case
```

Comment:
I think the 'why' would be good to add here, which I gather is that `ion_reader_read_lob_bytes` will return an error for a zero-length lob.

Reply:
I believe that to be a point in time though--that is a bug in Ion C, I can add a note about that though.